Iis2iclx sensor: move odr and range into dts

avisconti opened the pull request:
STM IIS2ICLX inclinometer sensor

Move odr and range properties from Kconfig to DT.
Moreover, a new yaml file has been created common to I2C and SPI bus.

Tested on x_nucleo_iks01a3 shield.

avisconti committed:
dts/bindings: iis2iclx: create a common st,iis2iclx-common.yaml

Create a common binding file that will be included by all bindings
handled by iis2iclx driver.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: iis2iclx: Move range Kconfig property into dts

Converts iis2iclx range options (500mg, 1g, 2g, 3g) from Kconfigs
to Device Tree.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti committed:
drivers/sensor: iis2iclx: Move odr Kconfig property into dts

Move iis2iclx odr options from Kconfigs to Device Tree.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

gmarull commented on drivers/sensor/iis2iclx/iis2iclx.c:
```diff
@@ -568,12 +564,12 @@ static int iis2iclx_init_chip(const struct device *dev)
 
-	if (iis2iclx_accel_set_fs_raw(dev,
-				     IIS2ICLX_DEFAULT_ACCEL_FULLSCALE) < 0) {
+	LOG_INF("range is %d", fs);
```
maybe `LOG_DBG`?

avisconti replied:
ok, I was infact in doubt. Seems more for debug I agree.
Thanks

gmarull commented on drivers/sensor/iis2iclx/iis2iclx.c:
```diff
@@ -571,8 +566,8 @@ static int iis2iclx_init_chip(const struct device *dev)
 
-	iis2iclx->accel_freq = iis2iclx_odr_to_freq_val(CONFIG_IIS2ICLX_ACCEL_ODR);
-	if (iis2iclx_accel_set_odr_raw(dev, CONFIG_IIS2ICLX_ACCEL_ODR) < 0) {
+	LOG_INF("odr is %d", odr);
```
same here

avisconti replied:
yes, acked! Thx

gmarull reviewed (approved):
LGTM, two minor comments

pabigot commented on dts/bindings/sensor/st,iis2iclx-common.yaml:
```diff
@@ -0,0 +1,22 @@
+        the signal that is presented to the driver.
+
+    int-pin:
+      type: int
```
It's out of the PR's intended scope, and the same problem occurs elsewhere so won't block on this, but:

In another PR please review all uses of similar properties in ST devicetree bindings and make this and the others you'll find an `enum` too, like `st,iis2dlpc`'s `drdy-int` property.

avisconti replied:
@pabigot 
Yes, that's a good point. I'll do it in a separate PR.
Thanks

pabigot reviewed (approved):
Fine as is; noting an unrelated global change that should also be done separately.

avisconti committed:
drivers/sensor: iis2iclx: modify int-pin type in DT

Change int-pin property type to enum, as it may only
accept 1 or 2 as possible values.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented:
@pabigot 
I saw that iis2iclx was the only sensor other than iis2dlpc to use int-pin property. So I added the commit in this PR itself. The previous commits are unchanged, and this stuff is planned for v2.6.0 , so it is not an issue if it is delayed.

pabigot commented on dts/bindings/sensor/st,iis2iclx-common.yaml:
```diff
@@ -11,15 +11,19 @@ properties:
         the signal that is presented to the driver.
 
-    int-pin:
+    drdy-int:
```
We need to be careful about this, as it breaks anybody's out-of-tree devicetree file that uses `int-pin` for this property.

Probably best to leave it as is, but if it's going to change both should be present, `int-pin` should be marked deprecated, and the driver should accept both until `int-pin` is removed.

If we don't have that documented as a policy somewhere, we should.  @galak @mbolivar-nordic

avisconti replied:
I saw in #31965 that properties with name changed was marked depracated. I can do the same in fact. Thanks for pointing this out.

pabigot reviewed (changes requested):
I want more eyes on the potential change to the devicetree content before saying how this should be done.

pabigot commented on drivers/sensor/iis2iclx/iis2iclx.c:
```diff
@@ -690,7 +690,7 @@ static int iis2iclx_init(const struct device *dev)
 		.irq_pin = DT_INST_GPIO_PIN(inst, drdy_gpios),		\
 		.irq_flags = DT_INST_GPIO_FLAGS(inst, drdy_gpios),	\
-		.int_pin = DT_INST_PROP(inst, int_pin)
+		.drdy_int = DT_INST_PROP(inst, drdy_int)
```
I think this should be something like 

    COND_CODE_1(DT_HAS_PROP(DT_INST(inst), drdy_int),
                (DT_INST_PROP(inst, drdy_int)),
                (DT_INST_PROP(inst, int_pin)))

otherwise we're still breaking people.  But that could be part of the discussion of how to handle devicetree deprecations.

avisconti replied:
Yes, correct. I did it too quickly. Sorry.

avisconti replied:
I'm actually planning to go back to original name, even if I think that drdy-int is a better name, but I'll do it next week. In any case I think it would be productive to understand what is the policy in such a case.

pabigot replied:
That's the fastest path to getting this in; #32225 should result in a documented policy.

avisconti replied:
> That's the fastest path to getting this in; #32225 should result in a documented policy.

Sure, just did it!

pabigot commented on dts/bindings/sensor/st,iis2iclx-common.yaml:
```diff
@@ -20,3 +20,15 @@ properties:
+        - 1 # 3g    (0.122 mg/LSB)
+        - 2 # 1g    (0.031 mg/LSB)
+        - 3 # 2g    (0.061 mg/LSB)
+
```
`git commit` normally complains about empty last lines, not sure why it didn't here.

pabigot replied:
Looks like it's corrected in the next commit.

pabigot reviewed (approved):
An intermediate commit has a whitespace issue but checkpatch doesn't see it and it gets cleaned up in the next commit, so this seems fine.  Thanks for your work on this.
